src/ApiClient/ServerClient.ts

```diff
@@ -268,8 +256,9 @@ export default new (class ServerClient extends ApiClient<IEvents> {
         LoginHooks.addHook(this.getServerInfo);
         this.on("purgeCache", () => {
             this._serverInfo = undefined;
-            if (CredentialsProvider.token) {
-                void LoginHooks.runHooks(CredentialsProvider.token);
+            const token = AuthController.getTokenUnsafe();
+            if (token) {
+                LoginHooks.runHooks(token);
```

Comment:
While it does need to check if theres a token, it doesn't need to pass it down to LoginHooks, can you remove that if you're going to remove all other instances of the token being passed down through the app